openpype/plugins/actions/open_file_explorer.py

```diff
@@ -0,0 +1,167 @@
+import os
+import platform
+import subprocess
+import getpass
+
+from openpype.client import (
+    get_project,
+    get_asset_by_name,
+)
+from openpype.pipeline import (
+    Anatomy,
+    LauncherAction,
+)
+
+
+class OpenTaskPath(LauncherAction):
+    name = "open_task_path"
+    label = "Explore here"
+    icon = "folder-open"
+    order = 500
+
+    def is_compatible(self, session):
+        """Return whether the action is compatible with the session"""
+        return "AVALON_ASSET" in session
+
+    def process(self, session, **kwargs):
+        from qtpy import QtCore, QtWidgets
+
+        project_name = session["AVALON_PROJECT"]
+        asset_name = session.get("AVALON_ASSET", None)
+        task_name = session.get("AVALON_TASK", None)
+        host_name = None    # never present in session
+
+        project = get_project(project_name)
+        asset = get_asset_by_name(project_name, asset_name)
+
+        data = self.get_workdir_data(project_doc=project,
+                                     asset_doc=asset,
+                                     task_name=task_name,
+                                     host_name=host_name)
+        anatomy = Anatomy(project_name)
+        result = anatomy.format(data)
+
+        # todo: implement custom template keys instead of 'work'
+        # Get template key
+        # template_key = get_workfile_template_key_from_context(
+        #    asset_name, task_name, host_name, project_name, dbcon
+        # )
+        workdir = result["work"]["folder"]
+
+        # Keep only the part of the path that was formatted by splitting up to
+        # the first stub - we can only explore up to there
+        STUB = "<<NULL>>"
+        valid_dir = workdir.split(STUB, 1)[0]
```

Comment:
This require `get_template_data` from `openpype.pipeline.template_data` and `get_workdir_data` can be removed.
```suggestion
        data = get_template_data(project_doc, asset_doc, task_name)

        anatomy = Anatomy(project_name)
        workdir = anatomy.templates["work"]["folder"].format(data)
        
        valid_dir = workdir.split("{", 1)[0]
```